### DISCUSSION TITLE
Linear elastic problem runs too slow

### DISCUSSION THREAD

**Discussion by Juns-Cheng** [idaholab/moose#29377] [Q&A Modules: Solid mechanics]
Hello, everyone. I am new to moose. I am testing the program with a very simple linear elastic problem: a spring with one end fixed and the other subjected to pressure (see attachment). There are 50k nodes in total. CALCULIX and ANSYS can solve the problem in seconds. But MOOSE converges very slow. 
I don't quite understand, because the problem is linear and needs no iteration at all. Any suggestions? 
[spring.zip](https://github.com/user-attachments/files/18051378/spring.zip)

**Comment by GiudGiud** [comment]
Hello

If you are solving a linear problem, you can use a linear solver (set the solve_type) to shave an extra nonlinear iteration.
If you are looking for performance, you will need to set the preconditioner. A common choice in solid mechanics is hypre boomeramg. you set it like this:

```
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
```

**Reply by GiudGiud** [reply]
I can't blindly download a zip on a work computer so if you need to share inputs please use a github repo

**Reply by Juns-Cheng** [reply]
Thanks for the suggestion. Now I use LINEAR solve type. The LU method works fine, but HYPRE does not converge. I will try tuning the solver parameters.
